### DISCUSSION TITLE
docker buildx push it's not working for insecure registry

### DISCUSSION THREAD

**Discussion by fjglira** [docker/buildx#2197] [Q&A]
Hi, I'm trying to use docker buildx --push to build for multi arch and push to a insecure https registry, this is my example:

`docker buildx create --name project-v4-builder --driver-opt network=host  --config buildkitd.toml`

The content of the config file it's something like this:
```
insecure-entitlements = [ "network.host", "security.insecure" ]
[registry."https://url_here"]
    http = false
    insecure = true
```

I run using this command:
`docker buildx build --push --platform=linux/arm64,linux/amd64,linux/s390x,linux/ppc64le --tag tag_here  -f Dockerfile.cross .`

But I'm getting `tls: failed to verify certificate: x509: certificate signed by unknown authority`. btw I also have the insecure registry configured in my daemon.json file

**Comment by marco74** [comment]
Sorry for being half a year later.

For official CA you can do this on debian (especially on debian-slim) by
```bash
apt-get update && apt-get -y install ca-certificates
```

In your config you mention `https`, `http = false` but `insecure = true`. This must be installed by copying your CA's certificate to `/usr/local/share/ca-certificates` and running `update-ca-certificates` then on each maching that wants to reach your registry. I did a similar setup, just published [this](https://gist.github.com/marco74/527880f4188dc304d14ec0435cacbb0c#file-selfsigned-md).
